ui/selectors/nft.ts

```diff
@@ -0,0 +1,55 @@
+import { NftContract } from '@metamask/assets-controllers';
+import { createSelector } from 'reselect';
+import { getMemoizedCurrentChainId } from '.';
+
+type NftState = {
+  metamask: {
+    allNftContracts: {
+      [account: string]: {
+        [chainId: string]: NftContract[];
+      };
+    };
+  };
+};
+
+function getNftContractsByChainByAccount(state: NftState) {
+  return state.metamask.allNftContracts ?? {};
+}
+
+export const getNftContractsByAddressByChain = createSelector(
+  getNftContractsByChainByAccount,
+  (nftContractsByChainByAccount) => {
+    const userAccounts = Object.keys(nftContractsByChainByAccount);
+
+    const allNftContracts = userAccounts
+      .map((account) =>
+        Object.keys(nftContractsByChainByAccount[account]).map((chainId) =>
+          nftContractsByChainByAccount[account][chainId].map((contract) => ({
+            ...contract,
+            chainId,
+          })),
+        ),
+      )
+      .flat()
+      .flat();
+
```

Comment:
The nested map calls can be simplified using flatMap to avoid the need for flat() calls.
Also, consider using Object.entries instead of Object.keys to directly access both the keys and values.

```suggestion
    const allNftContracts = Object.values(nftContractsByChainByAccount).flatMap(
      (chainIdToContracts) =>
        Object.entries(chainIdToContracts).flatMap(([chainId, contracts]) =>
          contracts.map((contract) => ({
            ...contract,
            chainId,
          })),
        ),
    );

```

Reply:
Love this! Only for the sake of time and QA, I'll merge as it is and update this in the next petname PR.